### ISSUE TITLE
Mopub Browser is crashing

### ISSUE THREAD

**Issue by ahmad-raza** [mopub/mopub-android-sdk#158]
MoPub browser crashes when you click on native ad & go to browser and click on cancel to go back.

See logs below:
Activity com.mopub.common.MoPubBrowser has leaked IntentReceiver android.widget.ZoomButtonsController$1@431b5aa8 that was originally registered here. Are you missing a call to unregisterReceiver()?

04-07 19:48:36.342: E/AndroidRuntime(12258): FATAL EXCEPTION: main
04-07 19:48:36.342: E/AndroidRuntime(12258): java.lang.IllegalArgumentException: Receiver not registered: android.widget.ZoomButtonsController$1@431b5aa8
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:654)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1165)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:378)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.widget.ZoomButtonsController.setVisible(ZoomButtonsController.java:405)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.widget.ZoomButtonsController$2.handleMessage(ZoomButtonsController.java:179)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.os.Handler.dispatchMessage(Handler.java:99)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.os.Looper.loop(Looper.java:213)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at android.app.ActivityThread.main(ActivityThread.java:4787)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at java.lang.reflect.Method.invokeNative(Native Method)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at java.lang.reflect.Method.invoke(Method.java:511)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:556)
04-07 19:48:36.342: E/AndroidRuntime(12258): 	at dalvik.system.NativeStart.main(Native Method)

**Comment by joeblubaugh** [comment]
Hi @ahmad-raza - 

Some more information should be able to help us reproduce. 

Are you seeing this regardless of the ad displayed? Do you have this problem with the sample ads?

When you click "cancel" are you clicking the X icon at the bottom of the screen? Are you clicking the Android "back" button?

What hardware are you testing on?

**Comment by ben-efiz** [comment]
Hey @joeblubaugh, we are also getting a lot of those crashes, too. It is our **biggest** crash group with **>3000** crashes! Taking a look at the play console i can precisely tell the date when it startet: **28th of January**, and it was **without** releasing a version in this period on our side or any mopub sdk update. So it might be advertisement related?

I am also having crashes during development and i can tell it happens shortly after banner ad was loaded and without clicking **anything** (not even the banner ad). The log says just before crashing

`<"mopub://finishLoad">`

I can provide the following information:

Android Version
Android 4.2 	59,6 %
Android 4.3 	8,2 %
Android 4.0.3 - 4.0.4 	16,3 %
Android 4.1 	15,4 %
Android 3.2 	0,5 %
...

Devices

Xperia L (C2105) 	14,0 %
BELTEC_BC400 	9,4 %
Xperia L (C2104) 	3,7 %
Xperia Miro (ST23i) 3,2 %
...

If you have further questions, please let me know.

**Comment by ben-efiz** [comment]
I also have crashes on my Nexus 5 with Android 5.0.2. So it's not just a Xperia, Android 4.x bug.

**Comment by joeblubaugh** [comment]
@ben-efiz thanks for the extra info. What versions of the MoPub SDK do you have released? Is there a particular version where you see these crashes more?

**Comment by ben-efiz** [comment]
We started with version 3.3.0 in December and then without any changes the
crashes appeared in the live version beginning from 28th of January.
February we updated to version 3.5.0 with no changes in frequency.
Am 10.04.2015 7:17 nachm. schrieb "Joe" <notifications@github.com>:

**Comment by joeblubaugh** [comment]
@ben-efiz can you email support@mopub.com and include this issue link? We'd like to investigate more.

**Comment by ben-efiz** [comment]
@joeblubaugh done.

**Comment by pablo-navarro** [comment]
Try with this on MopubBrowser:

```java

@Override
    protected void onDestroy() {
        super.onDestroy();
        mWebView.setVisibility(View.GONE);
        long timeout = ViewConfiguration.getZoomControlsTimeout();
        new Timer().schedule(new TimerTask() {
            @Override
            public void run() {
            	mWebView.destroy();
            	mWebView = null;
            }
        }, timeout);
    }

```

**Comment by shmeadyy** [comment]
Thanks for reporting this @ben-efiz.  We identified this as an issue with the ZoomButtonsController and have deployed a fix that will be covered in in the 3.7 release of the SDK.  Be sure to update your version for a more permanent solution.  Thanks!
